File: backend/src/main/kotlin/com/group7/artshare/entity/AccountInfo.kt
package com.group7.artshare.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonManagedReference
import lombok.Data
import java.util.*
import javax.persistence.*

@Data
@Entity
class AccountInfo(
    @Column
    var email: String,

    @Column(nullable = false)
    var username: String,

    @JsonIgnore
    @Column(nullable = false)
    private var password: String,
)
{
    @Id
    @GeneratedValue
    val id: Long = 0L

    @OneToOne(mappedBy = "accountInfo", cascade = [CascadeType.ALL])
    @JsonBackReference
    var registeredUser: RegisteredUser? = null

    @Column
    var name: String?=null

    @Column
    var surname: String?=null

    @Column
    var country: String?=null

    @Column
    var dateOfBirth: Date? = null

    @Column
    var profilePictureId: Long? = null

    @JsonIgnore
    fun getPassword(): String {
        return password
    }

}

File: android/lib/models/profile/account_info.dart
class AccountInfo {
  final String email;
  final String username;
  final int id;
  // final String? registered_user;
  final String? name;
  final String? surname;
  final String? country;
  DateTime? date_of_birth;
  final int? profile_picture_id;

  AccountInfo({
    required this.email,
    required this.username,
    required this.id,
    // this.registered_user,
    this.name,
    this.surname,
    this.country,
    this.date_of_birth,
    this.profile_picture_id,
  });


  factory AccountInfo.fromJson(Map<String, dynamic> json) {
    AccountInfo info = AccountInfo(
      id: json["id"],
      email: json["email"],
      username: json["username"],
      // registered_user: json["registeredUser"],
      name: json["name"],
      surname: json["surname"],
      country: json["country"],
      profile_picture_id: json["profilePictureId"],
    );

    if(json["dateOfBirth"] != null) {
      info.date_of_birth = DateTime.parse(json["dateOfBirth"]);
    }

    return info;
  }
}